prysm/io.py

```diff
@@ -1375,3 +1375,94 @@ def read_sigfit_rigidbody(file):
             'dR': dR
         }
     return out
+
+def read_codev_int(filename):
+    """Read the content of a CodeV INT (.int/.cvg) file.
+    
+    Parameters
+    ----------
+    file : path_like
+        path to a file
+    
+    Returns
+    -------
+    'dict'
+        dictionary with keys: phase, intensity, meta
+
+    """
+    with open(filename, 'rt') as f:
+        # skip headerlines starting with "!"
+        while True:
+            title = f.readline()
+            if not title.lstrip().startswith('!'):
+                break
+        # read parameter line
+        param = f.readline().split()
+        # parse parameters
+        if param[0] == 'GRD':
+            num_cols = int(param[1])
+            if param[2] == 'R':
+                #radial grid data
+                #not implemented
+                raise ValueError('Radial Grid Data INT-File type not supported.')
+            else:
+                num_rows = int(param[2])
+            datatype = param[3]    #SUR | WFR | FIL
+            #need to test now because different orders are allow
+            p=4
+            while p < len(param):
+                if param[p].upper() == 'WVL':
+                    wavelength = float(param[p+1])
+                if param[p].upper() == 'NMB':
+                    pass
+                if param[p].upper() == 'SSZ':
+                    scale_size = float(param[p+1])
+                if param[p].upper() == 'NDA':
+                    nda = int(param[p+1])
+                if param[p].upper() == 'XSC':
+                    x_scale = float(param[p+1])
+                p += 1    
+            #read grid data
+            phase=np.array([], dtype=config.precision)
```

Comment:
I would prefer if the function worked loosely as follows,

1.  Read the entire file into memory at the outset
2. Leave the code mostly intact that reads the top few lines
3. Replace the numpy append with something like this:

```python
txt = p.read_text()
low = txt.find('\n')
low = txt.find('\n', low+1)
low = txt.find('\n', low+1)
s = txt[low+1:]
a = np.fromstring(s, sep=' ', dtype=config.precision)
```

`txt` being whatever you called reading the entire file.  The sequence of finds are a way to skip the first few lines without using something like readline.  If you already know what "low" should be when `s = txt[low+1:]` is done, you don't need them.

Then with `a = np.fromstring`, the array is read by numpy in C (or other numpy-like).  This is about 5x faster:

Before: 123 ms ± 8.71 ms
After:   25.7 ms ± 142 µs

The stdev being so reduced (5x->1.5ms or so, not 150usec, 10x more gain there than avg time) is also indicative that it works the python garbage collector much less, so when using the function as part of a large, intensive program it will actually be proportionally faster ("Before" will slow the program down more than "AfteR").

Reply:
sorry -- to clarfy, the `p` in my snippet here is different; your `p` is a parameter index.  The `p` in my snippet is a path object.  The `txt` variable is just a string holding the entire contents of the file.

Reply:
I did a new commit where I made these changes. Slightly different .. ;-)